config/recipes/elastic-agent/fleet-ingress-setup.yaml

```diff
@@ -0,0 +1,357 @@
+    # as inside. See the comments in the Agent spec below for more details.
+    - hosts: ["fleet.example.com"]
+      secretName: fleet-server-acme 
+    - hosts: ["es.example.com"]
+      secretName: es-acme
+    - hosts: ["kb.example.com"]
+      secretName: kb-acme
+  rules:
+    - host: "kb.example.com"
+      http:
+        paths:
+        - path: "/"
+          pathType: Prefix
+          backend:
+            service:
+              name: kibana-kb-http
+              port:
+                number: 5601
+    - host: "es.example.com"
+      http:
+        paths:
+        - path: "/"
+          pathType: Prefix
+          backend:
+            service:
+              name: elasticsearch-es-http
+              port:
+                number: 9200
+    - host: "fleet.example.com"
+      http:
+        paths:
+          - path: "/"
+            pathType: Prefix
+            backend:
+              service:
+                name: fleet-server-agent-http
+                port:
+                  number: 8220
+---
+apiVersion: agent.k8s.elastic.co/v1alpha1
+kind: Agent
+metadata:
+  name: fleet-server
+spec:
+  version: 9.1.0
+  http:
+    # Configuring the same certificates used for the ingress here has the effect that 
+    # the CA certificate that is expected in ca.crt inside this secret is propagated to the agents
+    # and configured in the FLEET_CA environment variable. 
+    # Without this the agents would only trust the self-signed certificates generated by ECK.
+    tls:
+      certificate:
+        secretName: fleet-server-acme
+  kibanaRef:
+    name: kibana
+  elasticsearchRefs:
+  - name: elasticsearch
+  mode: fleet
+  fleetServerEnabled: true
+  policyID: eck-fleet-server
+  deployment:
+    replicas: 1
+    podTemplate:
+      spec:
+        containers:
+        - name: agent
+          env:
+          # Force Elastic Agent to bootstrap itself through the public Fleet Server URL
+          # We are asuming here the certificates configured above are only valid for the public URL.
+          - name: FLEET_URL
+            value: https://fleet.example.com:443
+        serviceAccountName: fleet-server
+        automountServiceAccountToken: true
+        securityContext:
+          runAsUser: 0
+---
+apiVersion: agent.k8s.elastic.co/v1alpha1
+kind: Agent
+metadata: 
+  name: elastic-agent
+spec:
+  config:
+    fleet:
+      enabled: true
+    providers.kubernetes:
+      add_resource_metadata:
+        deployment: true
+  version: 9.1.0
+  kibanaRef:
+    name: kibana
+  fleetServerRef:
+    name: fleet-server
+  mode: fleet
+  policyID: eck-agent
+  daemonSet:
+    podTemplate:
+      spec:
+        volumes:
+        - name: fleet-ca
+          secret:
+            secretName: fleet-server-acme
+        containers:
+        - name: agent
+          env:
+          # - name: FLEET_CA
+          #   value: /mnt/extra/ca.crt
```

Comment:
This does not seem to be require? (it works for me without it) It it can be useful let's add a comment to explain when this should be uncommented?

Reply:
Oh yes, that was a miss on my part. I will delete it.